base/third_party/go-wrapper/go-wrapper

```diff
@@ -89,11 +89,14 @@ case "$cmd" in
 		if [ -f "./source-context.json" ]; then
 			service="${GAE_SERVICE:-default}"
 			version="${GAE_VERSION:-default}"
-			set -x; exec go-cloud-debug \
+			set +e; set -x; go-cloud-debug \
 				-sourcecontext ./source-context.json \
 				-appmodule "${service}" \
 				-appversion "${version}" \
 				-- "$(which $goBin)" "$@"
+			if [ "$?" -eq 3 ]; then # fallback to running the binary itself if failed to get the application default credentials.
```

Comment:
3 seems like a somewhat common exit code.. maybe something higher for more obscure?